R/drop_upload.R

```diff
@@ -1,20 +1,24 @@
 
 
-#'Uploads a file to Dropbox using PUT semantics.
+
+
+#'Uploads a file to Dropbox.
+#'
+#'This function will allow you to write files of any size to Dropbox(even ones
+#'that cannot be read into memory) by uploading them in chunks.
 #'
-#' This function will allow you to write files of any size to Dropbox(even ones
-#' that cannot be read into memory) by uploading them in chunks.
 #'@param file Relative path to local file.
-#'@param  dest The relative path on Dropbox where the file should get uploaded.
-#'@param overwrite Default behavior (\code{TRUE}) is to overwrite files in the
-#'  destination. Set to \code{FALSE} to prevent this.
-#' @param encode The file encoding
-#'@param autorename This value, either true (default) or false, determines what
-#'  happens when there is a conflict. If true, the file being uploaded will be
-#'  automatically renamed to avoid the conflict. (For example, test.txt might be
-#'  automatically renamed to test (1).txt.) The new name can be obtained from
-#'  the returned metadata. If false, the call will fail with a 409 (Conflict).
-#'  response code.
+#'@param path The relative path on Dropbox where the file should get uploaded.
+#'@param mode - "add" - will not overwrite an existing file in case of a
```

Comment:
Yes, skipping that for now because it's not entirely clear to me what it is doing. Add makes sense (add a new file). So does overwrite. 

update doc says:
> update String(min_length=9, pattern="[0-9a-f]+") Overwrite if the given "rev" matches the existing file's "rev"
I don't fully understand how that's different from overwrite.

Reply:
I don't either, but smells a bit like the new args in `get_metadata`; can give filename, id, or revision